### PR TITLE
feat: add a resource for React

### PR DESCRIPTION
**Pull request by RitaDee**
## Assigned Issue

- [x] No issue assignment necessary.
- [ ] I am assigned to the related issue below.

## Description

This PR addresses https://github.com/shesharpnl/knowledge-hub/issues/39 by adding a resource for learning and mastering React.

## Type of PR

- [x] ✨ Feature
- [ ] 🐞 Bug fix
- [ ] 📄 Documentation update
- [ ] Others

### CODE REVIEW

File: resources/frontend-development.md
Comment (adiati98) [inline resources/frontend-development.md:18]: I like the description. However, so far, we don't have description for other resources. We might want to have consistency here for the future of our website.

I would love to have thoughts from @mariahlaqua and @saragerion on this matter.

Here are two proposals:

**1. Without description**:

```suggestion
- [React Course - Beginner's Tutorial for React JavaScript Library](https://www.youtube.com/watch?v=bMknfKXIFA8&t=853s)
```

**2. With description**:

```suggestion
- **[React Course - Beginner's Tutorial for React JavaScript Library](https://www.youtube.com/watch?v=bMknfKXIFA8&t=853s)**:

   This is a React beginner's course by Bob Ziroll for freeCodeCamp. You will learn React by building eight real-world projects and solving 140+ coding challenges in this course.
```

**Review by adiati98 — CHANGES_REQUESTED** [review]
Hey @RitaDee,
Thanks for your PR! 🙌

I left some reviews here, and would love to have some thoughts from other maintainers before we merge it in. 😊

File: resources/frontend-development.md
Comment (RitaDee) [reply resources/frontend-development.md:18]: I will wait for the opinion of the other maintainers. Thank you, @adiati98, for the review.

Comment (mariahlaqua) [reply resources/frontend-development.md:18]: Indeed we do not have a consistent practice with adding descriptions, and it is not stated as a requirement in the issue. I would say that we approve and merge the original pull request.

Then I would create a separate issue to improve the quality of resources by adding descriptions, and optionally screenshots or gifs. Once that is done we can update the original issue to include writing a description.

That's my thoughts @RitaDee @adiati98 @saragerion

Comment (adiati98) [reply resources/frontend-development.md:18]: Noted, @mariahlaqua. So, as we want to add description to resources moving forward — which can benefits readers to get a brief info about the resource — we'll include a description here. 👍

@RitaDee can you please make changes and go with suggestion no. 2? Thanks! 😊

Comment (saragerion) [reply resources/frontend-development.md:18]: Agree with @mariahlaqua :)

Comment (mariahlaqua) [reply resources/frontend-development.md:18]: I've made a new issue, https://github.com/shesharpnl/knowledge-hub/issues/53

regarding adding descriptions.

Comment (mariahlaqua) [inline resources/frontend-development.md:18]: @RitaDee I would like to see only the link to be a bit more informative with it's title, as we have freecodecamp linked elsewhere. 

I like the JavaScript library description and I liked the additional description for the course written by @adiati98 . As we have a new issue to handle descriptions I leave it up to @RitaDee to decide. Maybe you'd like to do more PRs ;)

Hope that makes sense! Thanks all, and thank you @RitaDee for your contribution and patience with our input. :)

**Review by mariahlaqua — CHANGES_REQUESTED** [review]

File: resources/frontend-development.md
Comment (RitaDee) [reply resources/frontend-development.md:18]: Thank you for the review, @mariahlaqua, @saragerion and @adiati98. I will make the changes you suggested.

Comment (RitaDee) [reply resources/frontend-development.md:18]: > @RitaDee I would like to see only the link to be a bit more informative with it's title, as we have freecodecamp linked elsewhere.
> 
> I like the JavaScript library description and I liked the additional description for the course written by @adiati98 . As we have a new issue to handle descriptions I leave it up to @RitaDee to decide. Maybe you'd like to do more PRs ;)
> 
> Hope that makes sense! Thanks all, and thank you @RitaDee for your contribution and patience with our input. :)

It does make sense, @mariahlaqua. I will create a PR for this in the new issue you created. Thank you.

File: resources/frontend-development.md
Comment (adiati98) [inline resources/frontend-development.md:12]: @RitaDee, because JavaScript has some frameworks and libraries, let's add an `h3` here as per [this suggestion](https://github.com/shesharpnl/knowledge-hub/pull/51#discussion_r1450079937) to give clarification. 👍

```suggestion
## JavaScript

### React
```

Comment (RitaDee) [reply resources/frontend-development.md:12]: Do you mean I should create a sub-title ```JavaScript Library``` in a h3 tag?

Comment (RitaDee) [reply resources/frontend-development.md:12]: > Just put "React" as an h3. 👍

Okay. Thank you.

**Review by mariahlaqua — APPROVED** [review]
Thanks @RitaDee !!

**Review by adiati98 — CHANGES_REQUESTED** [review]
@RitaDee thanks for the changes! 🙌

I left the final review here. Once it's fixed, this PR is ready to be merged.

I really like and appreciate your proactiveness; ask questions and the prompt fixes! ✨

For your future reference, you might also want to pay more attention to the code suggestions to save you time from going back and forth. 🙂

**Review by adiati98 — APPROVED** [review]
Thank you so much, @RitaDee! 🙌

Feel free to check other issues if you would like to contribute further!

git merge gif

**Pull request merged**
